### PR TITLE
Add 302 release notes

### LINKED ISSUE #125: Release notes for 302
## Dain Sundstrom       
- [x] all checked
-  01-31 15:18:24    10546eaa02 Reformat ParquetTester
-  01-31 15:18:24    bf01fcdc5e Add dynamic batch sizing in Parquet reader
-  01-31 15:27:30    99bd6ba914 Rename Context method to match field names
-  01-31 15:27:30    bacf59f6c5 Add physical* and network* metrics
-  01-31 15:27:30    e81eefea8f Remove unused methods
-  01-31 16:30:15    1fad470151 Disable ConnectorIdentity#equals and #hashCode
-  01-31 16:30:15    4a4bf3f288 Improve ConnectorIdentity#toString
-  02-04 14:45:12    6903d9b43e Add CLI option --trace-token to set the trace token #56.
-  02-04 16:55:23    f7fe35c2b3 Fix starvation while waiting for minimum number of workers
-  02-05 11:41:04    d6fb5e0f17 Add peak task user memory usage to query stats
-  02-05 12:02:33    d7d6166f88 Add spilled data to query stats, CLI, and Web UI
-  02-05 12:32:15    7ff3b40583 Allow function implementation choices to have different dependencies
-  02-06 09:06:30    275bb36af1 Remove system pool entirely
## David Phillips       
- [x] all checked
-  01-31 17:15:52    0b55a32373 Remove unused class ConnectorAnalyzeMetadata
-  02-02 01:31:23    3fc4affe62 Use class reference for BlockMissingException
-  02-02 16:17:58    02711a863a Move ScheduledSplit to execution package
-  02-02 16:17:58    6082254aa4 Move OutputBuffers to execution.buffer package
-  02-02 16:17:58    8ca92f2d4a Move TaskSource to execution package
-  02-02 16:37:56    d983340b51 Cleanup code in ORC module
-  02-04 14:26:47    81d77ca84b Fix grammar in error messages for CDF functions
-  02-04 15:30:50    20fb7d848c Document hive.recursive-directories
-  02-05 14:04:02    c2f23eb333 Rename StageExecutionStrategy to StageExecutionDescriptor
-  02-05 15:32:28    8ba4f26d8d Rename methods in StageExecutionDescriptor
-  02-05 17:04:29    30f7812492 Improve performance for positioned reads on S3
-  02-05 17:04:29    5c57d92342 Throw IOException for unrecoverable S3 errors
-  02-05 22:18:08    0a543f6f83 Add option not to delete target directory on failure
-  02-05 22:18:08    14cc065679 Remove redundant null check
-  02-06 11:07:04    4575f98086 Introduce DynamicConfigurationProvider
-  02-06 11:07:04    4daf53896b Add support for extra credential in JDBC driver
-  02-06 11:07:04    50224b3563 Add HiveGcsModule
-  02-06 11:07:04    70f733f2b3 Rename HdfsConfigurationUpdater to HdfsConfigurationInitializer
-  02-06 11:07:04    9a09af14f1 Add credentials in Identity and ConnectorIdentity
-  02-06 11:07:04    de7b51624c Add support for extra credential in Presto CLI
-  02-06 11:07:04    fa02de3ba7 Add header for extra credential
## Grzegorz Kokosiński 
- [x] all checked
-   02-01 02:13:30    1d625984d8 Remove PlanNodeMatcher
-   02-01 02:13:30    431e230405 Static import DEFAULT_MATCHER
-   02-01 02:13:30    4efe2808be Add context parameter to custom pattern matching functions
-   02-01 02:13:30    5d57936af0 Make Matcher to return the Stream of Match
-   02-01 02:13:30    6375f4c549 Remove Matcher, move match to Pattern
-   02-01 02:13:30    642330b5b0 Make Pattern#previous to be Optional
-   02-01 02:13:30    68e22ec17e Add missing requireNonNull to matching related constructors
-   02-01 02:13:30    70800a9365 Remove unused variable
-   02-01 02:13:30    87ed0f345e Remove value from Match
-   02-01 02:13:30    e0d02adc34 Inline Patterns implementations from DefaultMatcher
-   02-01 12:34:13    5d384b2f1d Add Elasticsearch connector
-   02-04 13:38:30    b59e5834a0 Make Git quiet when running tests in Travis
-   02-05 01:32:25    34858db37d Handle S3 empty directories as directories
## Karol Sobczak        
- [x] all checked
-  02-01 10:11:38    ca11249014 Add physical* and internalNetwork* stats to QueryStatistics
-  02-04 06:45:39    c3b113efa1 Do not reset configuration properties to default values
-  02-04 06:45:39    fce297a53f Use passed featuresConfig in LocalQueryRunner
-  02-04 09:00:10    1a867a639b Expose plan node stats and cost in event listener
-  02-04 09:00:10    552cf2a59a Document that some QueryStatistics field are not backward compatible
-  02-05 02:22:38    c3675ee9b3 Fix Javadoc for QueryStatistics#planNodeStatsAndCosts
-  02-06 04:41:11    0a9b8047f3 Revoke memory after initial output page has been produced in tests
-  02-06 04:41:11    1492b3f8de Return spill future directly
-  02-06 04:41:11    5fef5aa586 Allow memory revoke only during operator finish phase
-  02-06 04:41:11    79744f294a Convert revocable memory to user memory on hash aggregation finish
-  02-06 04:41:11    e69b668631 Produce more than single page in testHashAggregation
## Martin Traverso      
- [x] all checked
-  02-04 11:37:33    d5dfc99a1b Preserve quotedness in QualifiedName
-  02-04 16:46:08    7569071af2 Add approx_distinct for boolean
-  02-04 21:10:28    890a6a046d Clarify handling of constant properties
-  02-04 21:10:28    c9f8a2385f Propagate partition properties for full outer join
-  02-05 06:56:16    12c9b5086a Support boolean type in EXPLAIN IO
-  02-05 06:56:16    efc95d6ddb Rename classes to match coding style
-  02-05 23:24:22    7567baf65e Add SphericalGeography type
## Piotr Findeisen      
- [x] all checked
-  02-01 00:24:27    0c725747cb Remove TODO related to stats caching
-  02-01 00:24:27    363a3c388e Capture TaskManagerConfig on a field in LocalQueryRunner
-  02-01 00:24:27    dc1d66fb32 Add Exchange before GroupId to improve Partial Aggregation
-  02-01 03:24:50    eddafb528b Fix merge conflict
-  02-01 07:16:18    1116011e9f Rename VarcharType#getLengthSafe to #getBoundedLength
-  02-01 07:16:18    8af788806f Remove deprecated VarcharType#getLength method
-  02-01 07:16:58    27a1b0e304 Minor code fixes in AbstractTestParquetReader
-  02-01 07:16:58    d7210b7b40 Fix information_schema.columns queries for Kafka, Redis
-  02-02 13:10:02    ddf24210f8 Document note about LDAPS and JDK Upgrade
-  02-04 05:57:10    a8738947ee Remove redundant usages of StringBuilder
-  02-04 05:57:10    c290b17ada Import String.join statically
-  02-04 08:57:57    656be2c207 Make service principal hostname configurable in Kerberos auth
-  02-04 12:12:52    157ad3a766 Fix inconsistencies in hostname configuration property for Kerberos auth
-  02-05 06:14:47    1fa2cba1f3 Document RecordCursor#getReadTimeNanos
-  02-05 06:14:47    c48c8b672e Remove incorrect getReadTimeNanos() implementations
-  02-05 07:45:50    0ebbea6bf3 Check overlapping on Domain directly in TupleDomainParquetPredicate
-  02-05 07:45:50    1f4caa8e2c Make #getDomain return nonnull value
-  02-05 07:45:50    7296700cd6 Assume Domain from file cannot be isNone()
-  02-05 07:45:50    7d2f5a9299 Use known Presto Type instead of reconstructing
-  02-05 07:45:50    8e3bf91de2 Remove redundant `else`
-  02-05 07:45:50    972fdec09a Remove unused ParquetTypeUtils#getPrestoType method

### PR DESCRIPTION
**Pull request by dain**
Resolves #125

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-302.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+Redis Connector Changes
+-----------------------
+
+* Fix query failure when reading ``information_schema.columns`` without an equality condition on ``table_name``. (:issue:`120`)
+
+SPI Changes
+-----------
+
+* Include query peak task user memory in ``QueryCreatedEvent`` and ``QueryCompletedEvent``. (:issue:`163`)
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/release/release-302.rst]:
```
* Include physical and internal network input data size in ``QueryCompletedEvent``.
```

File: presto-docs/src/main/sphinx/release/release-302.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+---------------
+
+* Fix cluster starvation when wait for minimum number of workers is enabled. (:issue:`155`)
+* Fix backup of queries blocked waiting for minimum number of workers. (:issue:`155`)
+* Fix failure when preparing statements that contain a quoted reserved word as a table name. (:issue:`80`)
+* Fix query failure when spilling is triggered during certain phases of query execution. (:issue:`164`)
+* Add :doc:`/connector/elasticsearch`. (:issue:`118`)
+* Add support for ``boolean`` type to :func:`approx_distinct`. (:issue:`82`)
+* Add support for boolean columns to ``EXPLAIN`` with type ``IO``. (:issue:`157`)
+* Add SphericalGeography type and related functions. (:issue:`166`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-302.rst]: How about
```
* Add ``SphericalGeography`` type and related :doc:`geospatial functions </functions/geospatial>`. (:issue:`166`)

File: presto-docs/src/main/sphinx/release/release-302.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+* Remove deprecated system pool. (:issue:`168`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-302.rst]: "system memory pool"?

File: presto-docs/src/main/sphinx/release/release-302.rst [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+* Improve query performance for certain queries involving ``ROLLUP``. (:issue:`105`)
+* Preserve table name quoting in the output of ``SHOW CREATE VIEW``. (:issue:`80`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-302.rst]: This seems like a bug and thus should be above with the fixes

File: presto-docs/src/main/sphinx/release/release-302.rst [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+CLI Changes
+-----------
+
+* Add ``--trace-token`` option to set the trace token. (:issue:`117`)
+* Add spilled data size debug output. (:issue:`161`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-302.rst]: How about
```
* Display spilled data size as part of debug information. (:issue:`161`)

File: presto-docs/src/main/sphinx/release/release-302.rst [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+* Add support for client-provided extra credentials that can be utilized by connectors. (:issue:`124`)
+
+Hive Connector Changes
+----------------------
+
+* Fix Parquet predicate pushdown for ``smallint``, ``tinyint`` types. (:issue:`131`)
+* Add support for Google Cloud Storage (GCS). Credentials can be provided globally using the
+  ``hive.gcs.json-key-file-path`` configuration property, or as a client-provided extra credential
+  named ``hive.gcs`` if the ``hive.gcs.use-access-token`` is enabled. (:issue:`124`)
+* Fix support for tables with `external_location` property pointing to an empty S3 directory. (:issue:`75`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-302.rst]: Double backticks

Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-302.rst]: Maybe change to
```
* Allow creating tables with the ``external_location`` property pointing to an empty S3 directory. (:issue:`75`)

Comment (martint) [reply presto-docs/src/main/sphinx/release/release-302.rst]: We weren't sure whether the fix was about creating tables that look like that or running queries over such tables.

Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-302.rst]: It is for creating tables

> If you try to create a table with this S3 prefix you will get an
"External location must be a directory" error.

I also read the code

Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-302.rst]: The problem is we verify the location when creating the table and that check was failing due to not being a directory that exists.

File: presto-docs/src/main/sphinx/release/release-302.rst [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  for the Kerberos service principal. (:issue:`146`, :issue:`153`)
+* Add support for client-provided extra credentials that can be utilized by connectors. (:issue:`124`)
+
+Hive Connector Changes
+----------------------
+
+* Fix Parquet predicate pushdown for ``smallint``, ``tinyint`` types. (:issue:`131`)
+* Add support for Google Cloud Storage (GCS). Credentials can be provided globally using the
+  ``hive.gcs.json-key-file-path`` configuration property, or as a client-provided extra credential
+  named ``hive.gcs`` if the ``hive.gcs.use-access-token`` is enabled. (:issue:`124`)
```
[the remark this answers is not in the document]
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-302.rst]:
```
named ``hive.gcs.oauth`` if the ``hive.gcs.use-access-token`` is enabled. (:issue:`124`)

File: presto-docs/src/main/sphinx/release/release-302.rst [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  for the Kerberos service principal. (:issue:`146`, :issue:`153`)
+* Add support for client-provided extra credentials that can be utilized by connectors. (:issue:`124`)
+
+Hive Connector Changes
+----------------------
+
+* Fix Parquet predicate pushdown for ``smallint``, ``tinyint`` types. (:issue:`131`)
+* Add support for Google Cloud Storage (GCS). Credentials can be provided globally using the
+  ``hive.gcs.json-key-file-path`` configuration property, or as a client-provided extra credential
+  named ``hive.gcs.oauth`` if the ``hive.gcs.use-access-token`` is enabled. (:issue:`124`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-302.rst]:
```
  named ``hive.gcs.oauth`` if the ``hive.gcs.use-access-token`` configuration property is enabled. (:issue:`124`)

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**